json/src/main/scala/com/jpmc/json/Json.scala

```diff
@@ -0,0 +1,71 @@
+package com.jpmc.json
+
+sealed trait Json extends Product with Serializable
+
+object Json {
+  case object Null extends Json
+
+  final case class String(value: Predef.String) extends Json
+
+  final case class Int(value: scala.Int) extends Json
+
+  final case class Double(value: scala.Double) extends Json
+
+  final case class Bool(value: scala.Boolean) extends Json
+
+  sealed trait Doc extends Json
```

Comment:
It probably skipped your mind while coding, but I think it’s a good practice to use `sealed trait Doc extends Json with Product with Serializable` here. It stops namespace pollution and Nicolas had an example in one of his talks, I’ll paste a link here if I find it.